core/postsolve.gms

```diff
@@ -164,9 +164,9 @@ display s_actualbudgetco2;
 if(cm_iterative_target_adj eq 6,
 *JeS* Update tax levels/ multigasbudget values to reach the peak CO2 budget
  
-p_actualbudgetco2(t) =           sum(ttot$(ttot.val < t.val AND ttot.val > 2010), (sum(regi, (vm_emiTe.l(ttot,regi,"co2") + vm_emiCdr.l(ttot,regi,"co2") + vm_emiMac.l(ttot,regi,"co2"))) * sm_c_2_co2 * pm_ts(ttot)))
-                              + sum(regi, (vm_emiTe.l(t,regi,"co2") + vm_emiCdr.l(t,regi,"co2") + vm_emiMac.l(t,regi,"co2")))*sm_c_2_co2 * (pm_ts(t) * 0.5 + 0.5)
-                              + sum(regi, (vm_emiTe.l("2010",regi,"co2") + vm_emiCdr.l("2010",regi,"co2") + vm_emiMac.l("2010",regi,"co2")))*sm_c_2_co2 * 2;
+p_actualbudgetco2(t) = sum(ttot$(ttot.val < t.val AND ttot.val > 2020), (sum(regi, (vm_emiTe.l(ttot,regi,"co2") + vm_emiCdr.l(ttot,regi,"co2") + vm_emiMac.l(ttot,regi,"co2"))) * sm_c_2_co2 * pm_ts(ttot)))
+                       + sum(regi, (vm_emiTe.l(t,regi,"co2") + vm_emiCdr.l(t,regi,"co2") + vm_emiMac.l(t,regi,"co2")))*sm_c_2_co2 * (pm_ts(t) * 0.5 + 0.5) !! I think this half a year added here would yield incorrect computations if the time step is not 5 years
```

Comment:
If TEST_TS is set, we would get emissions from 2030 till t + [emissions from t] * (10/5 + 0.5) + [emissions from 2020] * 2. So we're missing 2.5 years. (I think)

Reply:
yes, that doesn't seem to make sense. I would use the same calculation as above.